### PR TITLE
GLTFExporter: Support keyframe animation, morph targets, and skinning

### PR DESCRIPTION
**Pull request by donmccurdy**
/cc @fernandojsg

### CODE REVIEW

**Comment by donmccurdy** [comment]
Should note that this allows export of keyframe animations and animated morph targets, but no skinning support yet.

**Comment by donmccurdy** [comment]
Ok, skinning support is also working now.

File: examples/js/exporters/GLTFExporter.js [hunk 1]
```diff
@@ -53,20 +60,26 @@ THREE.GLTFExporter.prototype = {
...
+			embedImages: true,
+			animations: []
 		};
 
 		options = Object.assign( {}, DEFAULT_OPTIONS, options );
 
+		if ( options.animations.length > 0 ) {
+
+			// Only TRS properties, and not matrices, may be targeted by animation.
+			options.trs = true;
```
Comment (fernandojsg) [inline examples/js/exporters/GLTFExporter.js]: Is this mandatory by the spec? Or is it just to ease the loader. I understand that the samplers for animation should be just `translate`, `rotate and `scale`, but if the node has a initial position defined by a matrix should that affect anyhow?

Comment (donmccurdy) [reply examples/js/exporters/GLTFExporter.js]: It's required by the spec, not sure why: https://github.com/KhronosGroup/glTF/blob/master/specification/2.0/README.md#transformations

Comment (fernandojsg) [reply examples/js/exporters/GLTFExporter.js]: Ah ok I didn't notice that comment. thanks. I'm also not sure why is needed, but that's another conversation :) Nice job here!

**Review by fernandojsg — APPROVED** [review]

**Comment by mrdoob** [comment]
Awesome stuff Don!

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**